packages/app/src/server/models/bookmark-folder.ts

```diff
@@ -0,0 +1,91 @@
+import { Ref, IUser } from '@growi/core';
+import ExtensibleCustomError from 'extensible-custom-error';
+import {
+  Types, Document, Model, Schema,
+} from 'mongoose';
+
+
+import loggerFactory from '../../utils/logger';
+import { getOrCreateModel } from '../util/mongoose-utils';
+
+const logger = loggerFactory('growi:models:bookmark-folder');
+
+export class InvalidParentBookmarkFolder extends ExtensibleCustomError {}
+
+
+export type IBookmarkFolderDocument = {
+  name: string
+  owner: Ref<IUser>
+  parent?: Ref<BookmarkFolderDocument>
+}
+export interface BookmarkFolderDocument extends Document {
+  _id: Types.ObjectId
+  name: string
+  owner: Types.ObjectId
+  parent?: Types.ObjectId | null
+}
+
+export interface BookmarkFolderModel extends Model<BookmarkFolderDocument>{
+  createByParameters(params: IBookmarkFolderDocument): IBookmarkFolderDocument
+  findParentFolderByUserId(user: Types.ObjectId | string): IBookmarkFolderDocument[]
+  findChildFolderById(parentBookmarkFolder: Types.ObjectId | string): Promise<IBookmarkFolderDocument[]>
+  deleteFolderAndChildren(bookmarkFolderId: string): void
+  updateBookmarkFolder(bookmarkFolderId: string, name: string, parent: string): BookmarkFolderDocument | null
+}
+
+const bookmarkFolderSchema = new Schema<BookmarkFolderDocument, BookmarkFolderModel>({
+  name: { type: String },
+  owner: { type: Schema.Types.ObjectId, ref: 'User', index: true },
+  parent: { type: Schema.Types.ObjectId, ref: 'BookmarkFolder', required: false },
+});
+
+
+bookmarkFolderSchema.statics.createByParameters = async function(params: IBookmarkFolderDocument): Promise<BookmarkFolderDocument> {
+  const { name, owner, parent } = params;
+  let bookmarkFolder;
+
+  if (parent === null) {
+    bookmarkFolder = await this.create({ name, owner, parent:  null }) as unknown as BookmarkFolderDocument;
+  }
+  else {
+    const parentFolder = await this.findById(parent);
+    if (!parentFolder) {
+      throw new InvalidParentBookmarkFolder("Parent folder doesn't exists");
+    }
+    bookmarkFolder = await this.create({ name, owner, parent:  parentFolder?._id }) as unknown as BookmarkFolderDocument;
+  }
```

Comment:
I think you are misunderstanding about Yuki San's FB.

`parent` variable passed from client side might be invalid value like `111` `aaa` not mongoose object id.
That's why we need to check the value is valid and it's exactly exist.